### PR TITLE
Find or initialize by 5.2.1

### LINKED ISSUE #31: 4.0.0 breaks find_or_initialize_by
This doesn't happen with 3.1.0. The attributes are not being merged into the new object.

The problem appears to be in `scope_for_create`. I'm not an expert in the internals of rails, but perhaps something like this might be safer?

```
    module ScopeForCreate
      def scope_for_create(attributes = nil)
        return super unless acting_as?
        scope = respond_to?(:values_for_create) ? values_for_create(attributes) : where_values_hash
        scope.merge!(where_values_hash(acting_as_model.table_name))
        scope.merge!(attributes) if attributes
        scope.merge(create_with_value)
      end
    end
```

### PR DESCRIPTION
**Pull request by grahamlyus**
Fixes #31

### CODE REVIEW

File: lib/active_record/acts_as/querying.rb [hunk 1]
```diff
@@ -28,8 +28,14 @@ def where!(opts, *rest)
 
     module ScopeForCreate
       def scope_for_create(attributes = nil)
+        unless acting_as?
+          return super(attributes) if Gem::Dependency.new('', '>= 5.2.1', '< 5.2.2').match?('', ActiveRecord.version)
```
Comment (manuelmeurer) [inline lib/active_record/acts_as/querying.rb]: Can you change this to

```
if Gem::Dependency.new('', '>= 5.2.1', '< 5.2.2').match?('', ActiveRecord.version)
  return super(attributes)
else
  return super()
end
```

Looks cleaner IMO.
Thanks!

**Comment by manuelmeurer** [comment]
Sorry for the delay...
Could you change the format to `if/else´ please?
Otherwise it looks good, thanks for working on this!

File: lib/active_record/acts_as/querying.rb (hunk 1 above)
Comment (grahamlyus) [reply lib/active_record/acts_as/querying.rb]: Sure, no problem, pushed

**Comment by manuelmeurer** [comment]
Great, thanks! v4.0.1 is released.

**Pull request merged**